packages/parser/src/pipeline/retail/segmentToCombat.ts

```diff
@@ -1,63 +1,231 @@
 import { pipe } from 'rxjs';
 import { filter, map } from 'rxjs/operators';
+import _ from 'lodash';
 
-import { CombatData, ICombatData, IMalformedCombatData } from '../../CombatData';
+import { CombatData, ICombatData, IMalformedCombatData, IShuffleCombatData, IShuffleRoundData } from '../../CombatData';
 import { ArenaMatchEnd } from '../../actions/ArenaMatchEnd';
-import { ArenaMatchStart } from '../../actions/ArenaMatchStart';
-import { ICombatEventSegment } from '../../types';
+import { ArenaMatchStart, ArenaMatchStartInfo } from '../../actions/ArenaMatchStart';
+import { CombatUnitType, ICombatEventSegment } from '../../types';
 import { computeCanonicalHash, nullthrows } from '../../utils';
 import { isNonNull } from '../common/utils';
 
+// Global buffer to hold recent shuffle rounds
+// once a shuffle-ending is detected this is reset
+let recentShuffleRoundsBuffer: IShuffleRoundData[] = [];
+let recentScoreboardBuffer: IShuffleRoundData['scoreboard'] = {};
+
+function roundsBelongToSameMatch(roundA: ArenaMatchStartInfo, roundB: ArenaMatchStartInfo) {
+  // ARENA_MATCH_START is identical for every round of a shuffle except the timestamp
+  if (roundA.bracket !== roundB.bracket) return false;
+  if (roundA.isRanked !== roundB.isRanked) return false;
+  if (roundA.item1 !== roundB.item1) return false;
```

Comment:
what is item1? 

does start info have combatants? i wonder if we should check for overlapping combatants as the best source of truth?

Reply:
item1 is unknown; startInfo doesnt have combatants (every field it has is used here except timestamp)